server/src/main/java/com/flipkart/grayskull/authz/GrayskullPermissionEvaluator.java

```diff
@@ -0,0 +1,36 @@
+package com.flipkart.grayskull.authz;
+
+import com.flipkart.grayskull.spi.GrayskullAuthorizationProvider;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.PermissionEvaluator;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.io.Serializable;
+
+/**
+ * A custom {@link PermissionEvaluator} that integrates Grayskull's authorization logic with Spring Security's
+ * method-level security. It acts as a bridge between the {@code @PreAuthorize("hasPermission(...)")} annotations
+ * and the {@link GrayskullAuthorizationProvider}.
+ */
+@Component
+@RequiredArgsConstructor
+public class GrayskullPermissionEvaluator implements PermissionEvaluator {
```

Comment:
we don't need to mention (or prefix) Grayskull everywhere. com.flipkart.graykull as group id should be enough I think.